plans/chew-large-datasets/test/ipfs_add_trickle_dag.go

```diff
@@ -3,39 +3,67 @@ package test
 import (
 	"context"
 	"fmt"
+	"os"
 
+	shell "github.com/ipfs/go-ipfs-api"
 	files "github.com/ipfs/go-ipfs-files"
 	coreopts "github.com/ipfs/interface-go-ipfs-core/options"
 	utils "github.com/ipfs/testground/plans/chew-large-datasets/utils"
+	"github.com/ipfs/testground/sdk/iptb"
 	"github.com/ipfs/testground/sdk/runtime"
 )
 
 // IpfsAddTrickleDag IPFS Add Trickle DAG Test
-func IpfsAddTrickleDag(runenv *runtime.RunEnv) {
+type IpfsAddTrickleDag struct{}
+
+func (t *IpfsAddTrickleDag) ExecuteAPI(runenv *runtime.RunEnv, cfg *utils.TestConfig) {
 	ctx, _ := context.WithCancel(context.Background())
 	ipfs, err := utils.CreateIpfsInstance(ctx, nil)
 	if err != nil {
 		panic(fmt.Errorf("failed to spawn ephemeral node: %s", err))
 	}
 
-	err = utils.ForEachCase(runenv, func(unixfsFile files.Node, isDir bool) error {
-		t := "file"
-		if isDir {
-			t = "directory"
-		}
-
+	err = cfg.ForEachUnixfs(runenv, func(unixfsFile files.Node, isDir bool) (string, error) {
 		addOptions := func(settings *coreopts.UnixfsAddSettings) error {
 			settings.Layout = coreopts.TrickleLayout
 			return nil
 		}
 
 		cidFile, err := ipfs.Unixfs().Add(ctx, unixfsFile, addOptions)
 		if err != nil {
-			return fmt.Errorf("Could not add %s: %s", t, err)
+			return "", err
 		}
 
-		fmt.Printf("Added %s to IPFS with CID %s\n", t, cidFile.String())
-		return nil
+		return cidFile.String(), nil
+	})
+
+	if err != nil {
+		runenv.Abort(err)
+		return
+	}
+
+	runenv.OK()
+}
+
+func (t *IpfsAddTrickleDag) ConfigureDaemon(runenv *runtime.RunEnv, spec *iptb.TestEnsembleSpec) {
+	spec.AddNodesDefaultConfig(iptb.NodeOpts{Initialize: true, Start: true}, "adder")
+}
+
+func (t *IpfsAddTrickleDag) ExecuteDaemon(runenv *runtime.RunEnv, ensemble *iptb.TestEnsemble, cfg *utils.TestConfig) {
+	node := ensemble.GetNode("adder")
+	client := node.Client()
+
+	err := cfg.ForEach(runenv, func(path string, file *os.File, isDir bool) (cid string, err error) {
+		if isDir {
+			// TODO
+			fmt.Println("Daemon API does not support adding trickle directories yet")
```

Comment:
There is no such thing as trickle directories :)

TrickleDag is about adding with a specific layout (today you can use either BalancedDag (default), TrickleDAg)